src/Catalog/PackageDeprecationItem.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Versioning;
+
+namespace NuGet.Services.Metadata.Catalog
+{
+    public class PackageDeprecationItem
+    {
+        public PackageDeprecationItem(
+            IReadOnlyCollection<string> reasons,
+            string message = null,
+            PackageDeprecationAlternatePackageItem alternatePackage = null)
+        {
+            if (reasons == null)
+            {
+                throw new ArgumentNullException(nameof(reasons));
+            }
+
+            if (!reasons.Any())
+            {
+                throw new ArgumentException(nameof(reasons));
+            }
+
+            Reasons = reasons;
+            Message = message;
+            AlternatePackage = alternatePackage;
+        }
+
+        public IReadOnlyCollection<string> Reasons { get; }
+        public string Message { get; }
+        public PackageDeprecationAlternatePackageItem AlternatePackage { get; }
+
+        public class PackageDeprecationAlternatePackageItem
+        {
+            public PackageDeprecationAlternatePackageItem(
+                string id)
+            {
+                Id = id ?? throw new ArgumentNullException(nameof(id));
+
+                // There is a bug in NuGet.Versioning preventing us from using VersionRange to calculate this.
+                // https://github.com/NuGet/Home/issues/6697
+                Version = "*";
+            }
+
+            public PackageDeprecationAlternatePackageItem(
+                string id,
+                string version)
+            {
+                Id = id ?? throw new ArgumentNullException(nameof(id));
+
+                if (version == null)
+                {
+                    throw new ArgumentNullException(nameof(version));
+                }
+
+                Version = new VersionRange(
+                    minVersion: NuGetVersion.Parse(version),
+                    maxVersion: NuGetVersion.Parse(version),
+                    includeMaxVersion: true).ToNormalizedString();
```

Comment:
I think we should be explicit about `includeMinVersion: true` for clarity.

Reply:
I will probably just make this accept a string.